apiserver/facades/agent/uniter/lxdprofile.go

```diff
@@ -227,21 +226,15 @@ func (u *LXDProfileAPI) RemoveUpgradeCharmProfileData(args params.Entities) (par
 			result.Results[i].Error = common.ServerError(common.ErrPerm)
 			continue
 		}
-		machine, err := u.getMachine(tag)
-		if err != nil {
-			result.Results[i].Error = common.ServerError(err)
-			continue
-		}
-		unit, err := u.getUnit(tag)
-		if err != nil {
-			result.Results[i].Error = common.ServerError(err)
-			continue
-		}
-		err = machine.RemoveUpgradeCharmProfileData(unit.Name())
+		_, err = u.getMachine(tag)
```

Comment:
Why continue to call this, are we attempting to find out if it's `not-provisioned` or `not-found`?

Reply:
This will be pulled out in pr fixing api/apiserver for uniter, was trying to break this into pieces.  Should have just returned empty results no errors.